components/constellation/constellation.rs

```diff
@@ -1458,7 +1459,8 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
 
     fn handle_script_loaded_about_blank_in_iframe_msg(&mut self,
                                                       load_info: IFrameLoadInfo,
-                                                      layout_sender: IpcSender<LayoutControlMsg>) {
+                                                      layout_sender: IpcSender<LayoutControlMsg>,
+                                                      doc_type: DocumentType) {
```

Comment:
Could we do it by having script send the constellation its immutable origin, which the constellation uses when deciding which script thread to use, then the script uses its parent's origin in the case that its url is about:blank? That way we'd avoid sending a mutable origin via the constellation, and non-initial about:blank would use (mostly) the same code path as regular loads.